src/engine/strat_engine/mdv.rs

```diff
@@ -80,7 +80,7 @@ impl<'a> Drop for MountedMDV<'a> {
 impl MetadataVol {
     /// Initialize a new Metadata Volume.
     pub fn initialize(pool_uuid: &PoolUuid, dev: LinearDev) -> EngineResult<MetadataVol> {
-        create_fs(dev.devnode().as_path())?;
+        create_fs(dev.devnode().as_path(), pool_uuid)?;
```

Comment:
This isn't really explained by commit message. Here we're initializing metadata volume w/ pool uuid. This seems different from the initializing of Stratis volumes.

Reply:
I updated issue #582 to discuss this topic.  If we agree the MVD should have the same UUID as the pool - I'll update the commit message.

Reply:
I think I've addressed these comments by moving the functions and changing the parameter type to &FilesystemUuid.